wgpu-hal/src/vulkan/device.rs

```diff
@@ -1414,7 +1414,11 @@ impl crate::Device for super::Device {
         //
         // https://github.com/gfx-rs/wgpu/issues/6867
         if let Some(label) = desc.label {
+            let l = self.shared.sampler_cache.lock();
+            // SAFETY: we are holding a lock on the sampler cache,
+            // so we can only be setting the name from one thread.
             unsafe { self.shared.set_object_name(raw, label) };
+            drop(l);
```

Comment:
Is this explicit drop here necessary? The lock will be dropped there anyway as it goes out of scope once the if-block exits. Or would `l` otherwise go unused?

Reply:
id rather be explicit, the drop shows that `l` is alive at least until the end of the scope: otherwise, the lock may be immediately discarded. Im not sure how NLL play with this, it might be fine, but i'd rather not rely on implicit behaviors if possible